### PR TITLE
*: some tiny improvements for the large transaction

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Some tiny improvements to make the large transaction work.

### What is changed and how it works?

1. Set the default txn size limit to 10G
2. Reduce memory footprint during insert operation for the large txn
3. Recycle the session buffer memory to avoid leak
4. Set rate limit to avoid 2PC prewrite overloads the TiKV
5. Start ttlManager for the large txn

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @coocood @jackysp

File: executor/insert_common.go [hunk 1]
```diff
@@ -408,6 +408,14 @@ func insertRowsFromSelect(ctx context.Context, base insertCommon) error {
 				if err = e.doBatchInsert(ctx); err != nil {
 					return err
 				}
+			} else if len(rows) > 300 {
```
Comment (jackysp) [inline executor/insert_common.go]: why 300?

Comment (tiancaiamao) [reply executor/insert_common.go]: Or any suggestion?

Comment (tiancaiamao) [reply executor/insert_common.go]: Done

File: config/config.toml.example [hunk 2]
```diff
@@ -201,7 +201,7 @@ bind-info-lease = "3s"
 # If using TiKV as the storage, the entry represents a key/value pair.
 # WARNING: Do not set the value too large, otherwise it will make a very large impact on the TiKV cluster.
 # Please adjust this configuration carefully.
-txn-total-size-limit = 104857600
+txn-total-size-limit = 10737418240
```
Comment (jackysp) [inline config/config.toml.example]: Please validate its value, it should not be greater than 10G.

File: store/tikv/2pc.go [hunk 3]
```diff
@@ -526,6 +535,13 @@ func (actionPrewrite) handleSingleBatch(c *twoPhaseCommitter, bo *Backoffer, bat
 		prewriteResp := resp.Resp.(*pb.PrewriteResponse)
 		keyErrs := prewriteResp.GetErrors()
 		if len(keyErrs) == 0 {
+			if bytes.Equal(c.primary(), batch.keys[0]) {
+				// After writing the primary key, if the size of the transaction is large than 4M,
+				// start the ttlManager. The ttlManager will be closed in tikvTxn.Commit().
+				if c.txnSize > 4*1024*1024 {
+					c.run(c, nil)
```
Comment (coocood) [inline store/tikv/2pc.go]: The ttlManager use `PessimisticLockTTL`, it assumes the primary key is pessimistic.

Comment (coocood) [reply store/tikv/2pc.go]: Need to pass `kill` pointer?

File: config/config.toml.example (hunk 2 above)
Comment (tiancaiamao) [reply config/config.toml.example]: Done

File: store/tikv/2pc.go (hunk 3 above)
Comment (tiancaiamao) [reply store/tikv/2pc.go]: It's not easy to pass that variable in the current code.
I'll tiny refactor the code later.

**Comment by coocood** [comment]
LGTM

**Comment by tiancaiamao** [comment]
The `DefTxnTotalSizeLimit` is set to 1G now.
Users can config it according to the memory of their hardware. @coocood @jackysp

**Pull request merged**